### DISCUSSION TITLE
Stop Feed

### DISCUSSION THREAD

**Discussion by Easygamer01s** [Crownicles/Crownicles#1945] [Nouvelles idées]
Bonjour à tous, 

Cette semaine j'ai été très étonné de voir qu'il était possible de monter rapidement dans le /top glory, sans spécialement avoir un niveau badass, un loot incroyable et une stratégie de jeu excellente... Il suffit d'avoir un serveur et un réseau de guildes pour se retrouver simplement entre tous petits. 

L'empire rouge respectent probablement les règles multicomptes, et personne n'est tenu à un quelconque code d'honneur en combat. 

Je propose cependant les modifications suivantes : 
- Les fights uniquement sur les serveurs arène du serveur
- Suppression de l'argument user dans la commande fight
- Impossible de lancer un fight si Energy/MaxEnergy < 0.80

Selon moi, ces modifications doivent venir ensemble pour qu'elles soient pertinentes..

Cordialement, 
Easy

**Comment by symsym-1629** [comment]
Bonjour easy, le feedfight est effectivement un gros problème présent dans draftbot, cependant tes propositions on aussi leur lots de désavantages :

Réserver les fights pour le serveur principal signifierait ôter le droit de fight aux personnes n'étant pas sur le serveur principal mais aussi forcer ceux qui souhaite fight, à rejoindre le serveur principal, ce qui pourrait être vu comme du "forcing" ou de la pub pour le serveur par certaines personnes. Mais le problème majeur à cette restriction est que cela créerait des inégalités plus fortes que le feedfignt, car les joueurs casuals qui joue à draftbot en "passe-temps" ne pourraient pas monter en ligue et se verraient le bonus de ligue ôté.

Secondement, je ne voie pas en quoi l'argument user permettrait de réduire le feedfight, car la personne a juste a accepter et puis c'est fini, ca revient à une demande de fight sans la mention, que le joueur peut rajouter dans un message séparé

Dernièrement, ce dernier point se trouve être assez intéressant, mais il reste restreint par la possibilité de se feed avec des comptes ayant un level plus bas, et donc une énergie max plus basse. la seule chose qui changerait serait le temps d'attente pour la personne qui se feedfight. 

Je tiens par ailleurs à noter que le système de gloire a été implémenté en partie pour palier à ce problème, car les personnes ayant un écart de gloire trop grand ne peuvent pas se fight.

Amicalement,

symsym

**Comment by romain22222** [comment]
Nous sommes conscient du problème, cependant, les solutions apportées au dessus ne vont pas, soit car elles amènent plus de problèmes qu'elles en résoudent, soit elles ne font que repousser le problème.

Une solution qui est dans les bacs actuellement est le #795 (matchmaking) mais demande un grand temps de développement et plus de joueurs.

Pour qu'une solution soit envisageable il faut que :
- Elle ne désavantage pas une personne vis à vis de si elle est sur le serveur ou non
- Elle **règle** le problème de feed-fight, pas juste le repousser
- Elle soit *relativement* facile à mettre en place
- Elle ne doit pas entraver l'amusement des joueurs

**Reply by BastLast** [reply]
Par contre empecher les fight sans être full énergie va être fait.

**Reply by Bombafox** [reply]
A ce moment là faudrait pouvoir acheter 3 fois max la regen de vie par jour et pas par semaine
Parce que je drop pas tant de popo d'énergie que ça et j'ai pas forcément envie d'attendre entre 9-12h pour pouvoir fight
Ça va juste ramener le problème avant les popo d'énergie si on attend d'être full life

**Reply by Bombafox** [reply]
Ou alors pouvoir fight si on a 1/4 ou 1/2 minimum de sa vie

**Reply by BastLast** [reply]
Ca sera probablement 80/90%

**Reply by Bombafox** [reply]
Du coup suffit juste de tous passer vétéran et on attend moin longtemps vu que c'est la classe qui a le moin d'énergie 🤭

**Comment by Easygamer01s** [comment]
On pourrait dédoubler la query sur le nombre de fights dans la semaine pour en faire une sur le joueur et l'autre sur la guilde ? 
Pour ainsi limiter le nombre de fights à 3 entre deux mêmes joueurs et 5 entre un joueur et les membres d'une même guilde. 

Je vois la commande getRankedFightsThisWeek dans logsReadRequests... 
ca suggère de récupérer l'id du player1 et l'id de guilde du player 2,
trouver le vecteur des id lié à la guilde du player2
et d'interroger le log sur base de ces critères.. 

Et de faire pareil en inversant les rôles..

Merci pour vos retours en tout cas !

**Comment by AyameKodaru** [comment]
Je m'incruste, vu que ça parle de nous 👀
Mais juste comme ça, limiter les adversaires en fonction du lvl et non de la gloire, c'est pas envisageable ? Parce qu'il est clair et net que sans feed un lvl 10 peut pas gagner contre un lvl 60. Pourtant ils peuvent fight : quelque soit le résultat c'est une forme de feed.
Et actuellement, en plus, il suffit d'abandonner pour donner la victoire à l'autre. Ce serait possible qu'un combat abandonné soit tout bonnement considéré annulé non ? De toute façon, quelqu'un qui abandonnerait volontairement pour empêcher les autres de gagner ne trouvera plus d'adversaires au bout d'un moment. Même si ça n'empêche pas tout le feedfight, à mon sens c'est des modifs raisonnables qui limiteraint grandement le feed.
Parce qu'on parle beaucoup de l'empire Rouge, et à raison puisqu'on l'exploite, mais on est loin d'être les seuls à feed pour autant, soyons honnêtes 😂

**Reply by symsym-1629** [reply]
pour la solution du niveau, faut pas oublier que le stuff fait aussi une différence

**Reply by Easygamer01s** [reply]
puis il reste des possibilités de saccager un match avec les attaques n'apportant pas de dégâts comme le soin..

**Reply by AyameKodaru** [reply]
Et alors ? Pour le coup c'est juste normal non ? Et ce serait pas limité par palier mais en fonction de chacun. Si t'as pas un bon stuff tu dois avoir ta chance contre quelqu'un qui a 10 lvl de moins que toi même s'il a un stuff de dingue. Et puis, l'idée même de la gloire, c'est bien de mettre en valeur ceux qui jouent le plus non ? Et ceux qui jouent bien ou valorisent les fight auront un bon stuff.

**Reply by AyameKodaru** [reply]
Et puis j'ai pas dit que c'était la solution ultime, je pense pas qu'elle existe, mais ça limiterait grandement le feed

**Reply by AyameKodaru** [reply]
> puis il reste des possibilités de saccager un match avec les attaques n'apportant pas de dégâts comme le soin..

J'ajouterai que, du reste, on a déjà donné des victoires à des joueuses qui ne pouvaient même pas nous mettre KO même si on commençait avec peu d'énergie et qu'on attaquait pas, parce que leurs stats sont tellement basses que ça ne suffisait pas. De plus toutes les classes n'ont pas d'attaques sans dégât, le paladin et le vétéran sont dans ce cas. Le mage aussi il me semble. Et le repos du tank et du chevalier sont importants, donc faut que l'adversaire ait une attaque honorable pour les mettre KO juste avec ça. Au final il n'y aurait que la protection du fantassin qui permettrait un feed facile.

**Reply by BastLast** [reply]
Bon je ferme cette issue car ça crée beaucoup de débat qui n'amène à rien, on va surveiller ça de pres et faire des ajustements nécessaires mais je pense que pr le moment l'équipe de dev est plus efficace pour travailler sur ce sujet qu'un thread random ou tout le monde propose pleins de trucs, ca va pas être un problème que l'on va resoudre rapidement mais on va se pencher dessus.
